File: src/Model/Message.php
<?php

namespace Model;


class Message {
    
    public $id;
    public $author;
    public $message;
    public $date;

    function __construct($id, $author, $message, \DateTime $date)
    {
        $this->id = $id;
        $this->author = $author;
        $this->message = $message;
        $this->date = $date;
    }

    public function setId($id)
    {
        $this->id = $id;
    }

    public function getId()
    {
        return $this->id;
    }

    public function setAuthor($author)
    {
        $this->author = $author;
    }

    public function getAuthor()
    {
        return $this->author;
    }

    public function setMessage($message)
    {
        $this->message = $message;
    }

    public function getMessage()
    {
        return $this->message;
    }

    public function setDate(\DateTime $date)
    {
        $this->date = $date;
    }

    public function getDate()
    {
        return $this->date;
    }

    public function isNew(){
        return (null === $this->id);
    }
}

File: Java/SocialChat/src/model/Message.java
package model;

public class Message {
	
	/** Id property */
	public int id;
	
	/** Author property */
	public String author;
	
	/** Message property */
	public String message;

	/** Construct
	 * 
	 * @param id
	 * @param author
	 * @param message
	 */
	public Message(int id, String author, String message) {
		super();
		this.id = id;
		this.author = author;
		this.message = message;
	}

	/** Retrieve id property
	 * 
	 * @return
	 */
	public int getId() {
		return id;
	}

	/** Set id property
	 * 
	 * @param id
	 */
	public void setId(int id) {
		this.id = id;
	}

	/** Retrieve author property
	 * 
	 * @return
	 */
	public String getAuthor() {
		return author;
	}

	/** Set author property
	 * 
	 * @param author
	 */
	public void setAuthor(String author) {
		this.author = author;
	}

	/** Retrieve message property
	 * 
	 * @return
	 */
	public String getMessage() {
		return message;
	}

	/** Set message property
	 * 
	 * @param message
	 */
	public void setMessage(String message) {
		this.message = message;
	}
}
